### PR TITLE
updates for python3 support in glue

### PR DESCRIPTION
**Pull request by ryandeivert**
*Description of changes:*

AWS Glue recently launched [support for python3](https://aws.amazon.com/about-aws/whats-new/2019/07/aws-glue-now-supports-ability-to-run-etl-jobs-apache-spark-243-with-python-3/)

Including updates to convert this project to python3. The changes include the following:
* update requirements for python3
* converting all package code to python3
* updating tests for python3
* bumping version (unsure what this should actually be) and updating makefile to get version correctly

*Testing:*

I ensured all tests are still passing using `make test`

- [x]  By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dacort** [comment]
@ryandeivert This is awesome, thank you! Been on my backlog for a bit, so I definitely appreciate the contribution.

Will take a look at this within the next week.

**Comment by juanqui** [comment]
> @ryandeivert This is awesome, thank you! Been on my backlog for a bit, so I definitely appreciate the contribution.
> 
> Will take a look at this within the next week.

Any updates on this?

**Comment by dacort** [comment]
Hi @juanqui - I unfortunately don't work at AWS anymore so not able to merge this in. 

I'll try to ping somebody on my old team there, but no guarantees. Feel free to fork this, though!

**Comment by sheyd** [comment]
I did a manual zip deploy of this and ran against S3 access logs using Glue 2.0 Python3 with `G.1X` nodes, and it worked successfully, both for initial run, and diff updates against latest partitions. @ryandeivert Thanks for putting this PR in and @dacort for putting this code together to begin with!

**Comment by serverlesspolska** [comment]
Any news on this PR? Why it has not been merged in?

**Comment by dacort** [comment]
@serverlesspolska I was the primary maintainer of this repo, but then left Amazon and wasn't able to find a replacement. That said, I'm back at Amazon now so will take a look at what needs to be done to update this / if we should continue maintaining it. A lot has changed in the past year. :)

File: Makefile [hunk 1]
```diff
@@ -58,7 +58,7 @@ CONVERTED_TABLE_NAME = $(call get_variable,CONVERTED_TABLE_NAME)
 S3_CONVERTED_TARGET = $(call get_variable,S3_CONVERTED_TARGET)
 S3_SOURCE_LOCATION = $(call get_variable,S3_SOURCE_LOCATION)
 
-require_service: 
+require_service:
 	@test -n "$(service)" || (echo "'service' variable must be defined" && exit 1)
 
 create_job: require_service
```
Comment (dacort) [inline Makefile:64]: This still creates a Python2 job so I'll need to tweak the Makefile to use the proper `glue-version` or `command` for Python3.

**Comment by dacort** [comment]
Thanks @ryandeivert for submitting this PR and I apologize for taking *checks notes* almost 2 years to merge it in. 

Keep an eye on the repo as there have been a lot of changes in Glue since then and Blueprints/Workflows may be the path forward.

**Pull request merged**